packages/cozy-harvest-lib/src/helpers/triggers.js

```diff
@@ -0,0 +1,85 @@
+import { randomDayTime } from './daytime'
+
+const DAILY = 'daily'
+const HOURLY = 'hourly'
+const MONTHLY = 'monthly'
+const WEEKLY = 'weekly'
+const VALID_FREQUENCIES = [DAILY, HOURLY, MONTHLY, WEEKLY]
+
+const DEFAULT_FREQUENCY = WEEKLY
+const DEFAULT_CRON = '0 0 0 * * 0' // Once a week, sunday at midnight
+// By default konnectors are run at random hour between 00:00PM and 05:00AM.
+const DEFAULT_TIME_INTERVAL = [0, 5]
+
+/**
+ * Stategies for building cron values
+ * See https://docs.cozy.io/en/cozy-stack/jobs/#cron-syntax
+ */
+const getCronStrategies = options => {
+  const { dayOfMonth, dayOfWeek, hours, minutes } = options
+  return {
+    [DAILY]: `0 ${minutes} ${hours} * * *`,
+    [HOURLY]: `0 ${minutes} * * * *`,
+    [MONTHLY]: `0 ${minutes} ${hours} ${dayOfMonth} * *`,
+    [WEEKLY]: `0 ${minutes} ${hours} * * ${dayOfWeek}`
+  }
+}
+
+/**
+ * Build a cron string for given konnector with given options
+ * See https://docs.cozy.io/en/cozy-stack/jobs/#cron-syntax
+ * @param  {string} frequency Frequency from `hourly`, `daily`, `weekly` or
+ * `monthly`.
+ * @param  {object} options   Object which may contain `hours`, `minutes` and
+ * `startDate`.
+ * @return {string}           The cron definition for trigger
+ */
+export const buildCron = (frequency, options = {}) => {
+  const sanitizedFrequency = VALID_FREQUENCIES.includes(frequency)
+    ? frequency
+    : DEFAULT_FREQUENCY
+
+  const sanitizedOptions = {
+    dayOfMonth: 1,
+    dayOfWeek: 1,
+    hours: 0,
+    minutes: 0,
+    ...options
```

Comment:
What other options will work here? Maybe we should have an exhaustive list of supported options instead of providing all

Reply:
Only those four options are working. Comment above needs to be updated.